docs/reference/snapshot-restore/restore-snapshot.asciidoc

```diff
@@ -1,187 +1,523 @@
+////
+[source,console]
+----
+POST _watcher/_start
+----
+// TEST[continued]
+////
+--
 
-In addition to entire data streams, you can restore only specific backing
-indices from a snapshot. However, restored backing indices are not automatically
-added to any existing data streams. For example, if only the
-`.ds-logs-2099.03.08-000003` backing index is restored from a snapshot, it is
-not automatically added to the existing `logs` data stream.
+. If you use {es} security features, log in to a node host, navigate to the {es}
+installation directory, and add a user with the `superuser` role to the file
+realm using the <<users-command,`elasticsearch-users`>> tool.
++
+For example, the following command creates a user named `restore_user`.
++
+[source,sh]
+----
+./bin/elasticsearch-users useradd restore_user -p my_password -r superuser
+----
++
+Use this file realm user to authenticate requests until the restore operation is
+complete.
 
-[discrete]
-=== Partial restore
+. Use the <<cluster-update-settings,update cluster settings API>> to set
+<<action-destructive-requires-name,`action.destructive_requires_name`>> to
+`true`. This lets you delete indices and data streams using wildcards.
++
+[source,console]
+----
+PUT _cluster/settings
+{
+  "persistent": {
+    "action.destructive_requires_name": false
+  }
+}
+----
+// TEST[setup:setup-snapshots]
 
-By default, the entire restore operation will fail if one or more indices or backing indices participating in the operation don't have
-snapshots of all shards available. It can occur if some shards failed to snapshot for example. It is still possible to
-restore such indices by setting `partial` to `true`. Please note, that only successfully snapshotted shards will be
-restored in this case and all missing shards will be recreated empty.
+. Delete all existing indices and data streams on the cluster, including all
+system indices. If you use {es} security features, use the previously created
+file realm user to authenticate the request.
++
+[source,console]
+----
+# Delete all indices
+DELETE *
 
+# Delete all data streams
+DELETE _data_stream/*
+----
+// TEST[continued]
 
-[discrete]
-[[change-index-settings-during-restore]]
-=== Changing index settings during restore
+. Restore the entire snapshot, including the cluster state. This also restores
+any system indices in the snapshot.
++
+--
+[source,console]
+----
+POST _snapshot/my_repository/my_snapshot_2099.05.06/_restore 
+{
+  "indices": "*",
+  "include_global_state": true
+}
+----
+// TEST[continued]
+// TEST[s/_restore/_restore?wait_for_completion=true/]
 
-Use the <<restore-snapshot-api-index-settings,`index_settings`>> parameter
-to override index settings during the restore process. For example, the
-following request will restore the index `index_1` without creating any
-replicas while switching back to the default refresh interval:
+If you take dedicated snapshots of the `.security` system index, restore it as well.
```

Comment:
Is this a thing people do? Do we recommend doing it elsewhere? If not I'd suggest leaving out this step. It's certainly true (and hopefully obvious) that if you don't have a full snapshot of the whole cluster then you'll have to restore it in pieces, but it's not the common case and I think it'll cause confusion to mention it here.

Reply:
We do recommend this here:

https://www.elastic.co/guide/en/elasticsearch/reference/master/security-backup.html#backup-security-index-configuration

I also confirmed with the @elastic/es-security team that we do still recommend dedicated security index backups in some cases.

However, I think you're right. if you do dedicated backups, it's clear that you need to restore those indices separately.
I'll remove this for now.